src/test/java/uk/gov/ons/ctp/integration/rhsvc/message/impl/UacEventReceiverImplUnit_Test.java

```diff
@@ -118,4 +132,41 @@ public void shouldFilterUacEventPrefix14() {
   public void shouldAcceptUacCreatedEvent() {
     acceptUacEvent(RespondentHomeFixture.QID_01, EventTopic.UAC_UPDATE);
   }
+
+  @Test
+  public void shouldRejectUacWhenPrerequisiteEventsDoNotExistInFirestore() throws CTPException {
+    when(acceptableEventFilter.filterAcceptedEvents(any(), any(), any(), any())).thenReturn(false);
+    prepareAndAcceptEvent(RespondentHomeFixture.QID_01, EventTopic.UAC_UPDATE);
+    verify(mockRespondentDataRepo, never()).writeUAC(uacFixture);
+  }
+
+  @Test
+  public void testExceptionThrown() throws CTPException {
+    UacEvent uacEvent = createUAC(RespondentHomeFixture.QID_01, EventTopic.UAC_UPDATE);
+    when(acceptableEventFilter.filterAcceptedEvents(any(), any(), any(), any())).thenReturn(true);
+    doThrow(new CTPException(CTPException.Fault.SYSTEM_ERROR)).when(mockRespondentDataRepo).writeUAC(uacEvent.getPayload().getUacUpdate());
+
+    CTPException thrown =
+        assertThrows(CTPException.class, () -> target.acceptUACEvent(uacEvent));
+
+    assertEquals(CTPException.Fault.SYSTEM_ERROR, thrown.getFault());
+    assertEquals("Non Specific Error", thrown.getMessage());
+  }
+
+  private UacEvent createUAC(String qid, EventTopic topic) {
```

Comment:
there is a ready made JSON file that you could load with `FixtureHelper` for this

Reply:
Done